components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1576,4 +1590,51 @@ private OAuthProblemException buildOAuthProblemException(AuthenticationResult au
             return OAuthProblemException.error(errorCode, errorMessage);
         }
     }
+
+    private void storeSidClaim(String redirectURL, OIDCSessionState sessionState){
+        if (redirectURL != null) {
+            if (redirectURL.contains("id_token")) {
+                byte[] decodedBytes = null;
+                if (redirectURL.contains("access_token")) {
+                    // Extracting id_token for response_type=id_token token.
+                    String[] idtoken = redirectURL.split("=")[2].split("&")[0].split("\\.");
+                    decodedBytes = Base64.decodeBase64(idtoken[1]);
+                } else {
+                    // Extracting id_token for response_type=id_token.
+                    String[] idtoken = redirectURL.split("=")[1].split("\\.");
+                    decodedBytes = Base64.decodeBase64(idtoken[1]);
+                }
+                String idToken = new String(decodedBytes);
+                JSONObject token = new JSONObject(idToken);
+                if (token.has("sid")) {
+                    String sid = token.getString("sid");
+                    if (sid != null) {
+                        sessionState.setSidClaim(sid);
+                    }
+                }
+            } else if (redirectURL.contains("code")) {
+                String sid = sessionState.getSidClaim();
+                String code = redirectURL.split("=")[1];
+                if (sid == null) {
+                    // Generating sid claim for authorization code flow.
+                    sid = UUID.randomUUID().toString();
+                    sessionState.setSidClaim(sid);
+                }
+                cacheSessionID(sid, code);
+            }
+        }
+    }
+
+    /**
+     * Store Authorization Code and SessionID for back-channel logout in the cache.
+     *
+     * @param sessionId
+     * @param authorizationCode
+     */
+    private void cacheSessionID(String sessionId, String authorizationCode) {
```

Comment:
Change the parameter order.

`private void cacheSessionID(String authorizationCode, String sessionId) {`

so it will accept paramters in key,value order